### PR TITLE
Release/v0.2.0

### PR DESCRIPTION
**Pull request by Jaime-Burgos**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,21 +4,26 @@ System handle to connect [*SOSS*][soss] to [*FIWARE*][fiware]
...
+$ git clone git@github.com:eProsima/SOSS-FIWARE.git src/soss-fiware
+$ colcon build --packages-up-to soss-fiware
+```
 ## Use case - Connecting with ROS2
 
-0. Prerequisites: curlpp and asio installed
+0. Prerequisites: [curlpp](https://github.com/jpbarrette/curlpp) and [asio](https://think-async.com/) installed
 1. [Create a colcon workspace](https://index.ros.org/doc/ros2/Tutorials/Colcon-Tutorial/#create-a-workspace).
+    ```
+    mkdir -p soss_wp/src
```
Comment (lemunozm) [inline README.md]:
```suggestion
    $ mkdir -p soss_wp/src
```

File: README.md [hunk 2]
```diff
@@ -4,21 +4,26 @@ System handle to connect [*SOSS*][soss] to [*FIWARE*][fiware]
...
+    cd soss_wp
```
Comment (lemunozm) [inline README.md]:
```suggestion
    $ cd soss_wp
```

File: README.md [hunk 3]
```diff
@@ -4,21 +4,26 @@ System handle to connect [*SOSS*][soss] to [*FIWARE*][fiware]
...
+    ```
 2. Clone the soss project into the source subfolder.
     ```
-    git clone git@github.com:osrf/soss_v2.git
+    git clone git@github.com:osrf/soss_v2.git src/soss
```
Comment (lemunozm) [inline README.md]:
```suggestion
    $ git clone git@github.com:osrf/soss_v2.git src/soss
```

File: README.md [hunk 4]
```diff
@@ -4,21 +4,26 @@ System handle to connect [*SOSS*][soss] to [*FIWARE*][fiware]
...
     ```
 3. Clone this project into the subfolder.
     ```
-    git clone git@github.com:eProsima/SOSS-FIWARE.git
+    git clone git@github.com:eProsima/SOSS-FIWARE.git src/soss-fiware
```
Comment (lemunozm) [inline README.md]:
```suggestion
    $ git clone git@github.com:eProsima/SOSS-FIWARE.git src/soss-fiware
```

File: README.md [hunk 5]
```diff
@@ -34,8 +39,14 @@ To install this package into a workspace already containing SOSS, just clone thi
                     └── fiware-test (soss-fiware-test colcon pkg)
     ```
 
-5. In the workspace folder, execute colcon: `colcon build --packages-up-to soss-fiware`.
-6. Source the current environment: `source install/local_setup.bash`.
+5. In the workspace folder, execute colcon: 
+    ```
+    colcon build --packages-up-to soss-fiware
```
Comment (lemunozm) [inline README.md]:
```suggestion
    $ colcon build --packages-up-to soss-fiware
```

File: README.md [hunk 6]
```diff
@@ -34,8 +39,14 @@ To install this package into a workspace already containing SOSS, just clone thi
...
+    ```
+6. Source the current environment:
+    ```
+    source install/local_setup.bash
```
Comment (lemunozm) [inline README.md]:
```suggestion
    $ source install/local_setup.bash
```

File: README.md [hunk 7]
```diff
@@ -72,18 +83,21 @@ Notice that this system handle maps soss messages dyrectly to a JSON compatible
...
 ### More information
 
-- For more information, you can see the [demo steps](fiware/sample/demo.md),
+- For more information, you can see the [demo steps](fiware/doc/demo.md),
 and the related [video](https://drive.google.com/open?id=1w90DAPkovjwj7673d5RfOINlAAc7kWb1)
 - Also, you can have a look to the [internal design](fiware/doc/design.md)
-- For a fast configuration, you can use the [dockerfile](Dockerfile)
+- For a fast configuration, you can use the [dockerfile](Dockerfile). 
+**NOTICE**: Both when building and running the Dockerfile, you must specify the option `--network=host`, 
+because FIWARE SH tells Orion context broker its own IP in order to make subscriptions,
```
Comment (lemunozm) [inline README.md]: The user maybe does not understands what is `SH`, maybe `SOSS FIWARE plugin` instead?

File: README.md [hunk 8]
```diff
@@ -72,18 +83,21 @@ Notice that this system handle maps soss messages dyrectly to a JSON compatible
...
+and for it to work the docker IP must be the same as the host IP.
```
Comment (lemunozm) [inline README.md]: The exact problem is that the docker IP is not accessible outside your computer. You could get another workaround, as routing the ip or aliasing it or putting fiware onion server in the same computer where soss is. In fact, we continue having this problem if the onion is outside of our LAN.
So maybe, I would put something as: 

*If fiware onion context broker can not reach the docker ip, you can run (or build if necessary) the docker with `--network=host` to share the network interface and make it accessible to your LAN.*

File: README.md [hunk 9]
```diff
@@ -72,18 +83,21 @@ Notice that this system handle maps soss messages dyrectly to a JSON compatible
...
 
 ### Changelog
 
-#### v0.1.2
-
+#### v0.2.0
 - Added dockerfiles
 - Added integration tests
 - Removed asio as local thirdparty
+- Subscription host and port automatically generated.
```
Comment (lemunozm) [inline README.md]:
```suggestion
- Subscription host and port automatically generated if they are not specified.
```

File: fiware/doc/demo.md [hunk 10]
```diff
@@ -1,20 +1,15 @@
-# Fiware system handle demo (April 2019)
-![](simple_diagram.png "Console diagram")
+# Fiware system handle demo
 
-## Demo steps
-> **print** after the sentence means that the message must be printed in video post-production
+[This link](https://drive.google.com/open?id=1w90DAPkovjwj7673d5RfOINlAAc7kWb1) contains a demonstration on how FIWARE can exchange messages with ROS2 using SOSS and the soss-fiware and soss-ros2 system handles.
```
Comment (lemunozm) [inline fiware/doc/demo.md]: I would change:
```suggestion
[This link](https://drive.google.com/open?id=1w90DAPkovjwj7673d5RfOINlAAc7kWb1) contains a demonstration on how FIWARE can exchange messages with ROS2 using SOSS through `soss-fiware` and `soss-ros2` system handles.
```

File: fiware/doc/demo.md [hunk 11]
```diff
@@ -194,5 +180,4 @@ curl 192.168.1.59:1026/v2/entities/hello_ros2?type=std_msgsString -s -S -H 'Acce
 curl 192.168.1.59:1026/v2/subscriptions/5cc3f909efccc70b574a4611 -s -S -H 'Accept: application/json' -X DELETE
 ```
 
-For removing all subscriptions, please, use the script found into utils `erase_subscriptions.sh`
-
+For removing all subscriptions, please, use the script found in fiware/utils called `erase_subscriptions.sh`
```
Comment (lemunozm) [inline fiware/doc/demo.md]: Add: 
**CAREFUL**, this script will delete all your instances in your server.

In fact, I do not know if we should distribute this script, is too dangerous, and a dds-fiware user does not want to erase all his data. The curl part for removing this concrete subscription could be enough.

**Review by lemunozm — DISMISSED** [review]

File: fiware/doc/demo.md (hunk 11 above)
Comment (Jaime-Burgos) [reply fiware/doc/demo.md]: Ok I'm removing it

File: README.md (hunk 7 above)
Comment (Jaime-Burgos) [reply README.md]: It's not necessary, this line has been deleted

File: README.md (hunk 8 above)
Comment (Jaime-Burgos) [reply README.md]: Changed

**Review by lemunozm — APPROVED** [review]

**Pull request merged**